server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDeleteIndexService.java

```diff
@@ -48,34 +45,37 @@ public class MetadataDeleteIndexService {
     private final Settings settings;
     private final ClusterService clusterService;
 
-    private final AllocationService allocationService;
+    private final ClusterStateTaskExecutor<DeleteIndexClusterStateUpdateRequest> executor;
 
     @Inject
     public MetadataDeleteIndexService(Settings settings, ClusterService clusterService, AllocationService allocationService) {
         this.settings = settings;
         this.clusterService = clusterService;
-        this.allocationService = allocationService;
+        executor = batchExecutionContext -> {
+            ClusterState state = batchExecutionContext.initialState();
+            for (ClusterStateTaskExecutor.TaskContext<DeleteIndexClusterStateUpdateRequest> taskContext : batchExecutionContext
+                .taskContexts()) {
+                try {
+                    state = deleteIndices(state, Sets.newHashSet(taskContext.getTask().indices()));
+                    taskContext.success(taskContext.getTask());
+                } catch (Exception e) {
+                    taskContext.onFailure(e);
+                }
+            }
+            if (state == batchExecutionContext.initialState()) {
+                return state;
+            }
+            return allocationService.reroute(state, "deleted indices");
```

Comment:
I believe in other places such as AutoCreateAction we do:
```suggestion
                    try (var ignored = batchExecutionContext.dropHeadersContext()) {
                        return allocationService.reroute(state, "deleted indices");
                    }
```

Reply:
Thanks, applied :)